registry/registry.go

```diff
@@ -345,36 +345,30 @@ func configureReporting(app *handlers.App) http.Handler {
 func configureLogging(ctx context.Context, config *configuration.Configuration) (context.Context, error) {
 	logrus.SetLevel(logLevel(config.Log.Level))
 
-	formatter := config.Log.Formatter
-	if formatter == "" {
-		formatter = "text" // default formatter
-	}
-
-	switch formatter {
+	switch config.Log.Formatter {
 	case "json":
 		logrus.SetFormatter(&logrus.JSONFormatter{
 			TimestampFormat:   time.RFC3339Nano,
 			DisableHTMLEscape: true,
 		})
-	case "text":
+	case "text", "":
+		// Set the config to the default ("text") formatter in case none was
+		// specified.
+		config.Log.Formatter = "text"
 		logrus.SetFormatter(&logrus.TextFormatter{
 			TimestampFormat: time.RFC3339Nano,
 		})
 	case "logstash":
 		logrus.SetFormatter(&logstash.LogstashFormatter{
-			Formatter: &logrus.JSONFormatter{TimestampFormat: time.RFC3339Nano},
+			Formatter: &logrus.JSONFormatter{
+				TimestampFormat: time.RFC3339Nano,
+			},
 		})
 	default:
-		// just let the library use default on empty string.
-		if config.Log.Formatter != "" {
-			return ctx, fmt.Errorf("unsupported logging formatter: %q", config.Log.Formatter)
-		}
-	}
-
-	if config.Log.Formatter != "" {
-		logrus.Debugf("using %q logging formatter", config.Log.Formatter)
```

Comment:
.. because I think the "bug" was actually here; this block should've used the local variable (`formatter`), not the config.